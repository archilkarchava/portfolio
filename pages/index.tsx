import Layout from '../components/Layout'

export const Home: React.FC = () => {
  return (
    <Layout className="flex flex-col items-center justify-center h-full">
      <div className="max-w-4xl p-10">
        <div>
          <h1 className="text-3xl font-bold leading-tight sm:text-5xl md:text-6xl">
            Archil Karchava
          </h1>
          <h2 className="text-xl font-semibold sm:text-3xl md:text-4xl">
            Software engineer
          </h2>
          <div className="my-6 text-lg sm:text-xl md:text-2xl">
            <h3 className="font-semibold">Technologies I use:</h3>
            <p>
              React • Redux • TypeScript • JavaScript • HTML • CSS • SASS •
              Webpack • REST • GraphQL • Go • Node.js • PostgreSQL
            </p>
          </div>
        </div>
        <footer className="flex items-center">
          <a
            href="mailto:hello@archil.dev"
            className="mr-3 text-xl md:mr-4 md:text-2xl"
          >
            hello@archil.dev
          </a>{' '}
          <a
            href="https://github.com/archilkarchava"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img src="/github.svg" alt="Github Logo" className="h-5 md:h-7" />
          </a>
        </footer>
      </div>
    </Layout>
  )
}

export default Home
