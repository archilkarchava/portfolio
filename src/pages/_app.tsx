import { FULL_NAME } from '@/lib/constants'
import '@/styles/index.css'
import type { AppProps } from 'next/app'
import Head from 'next/head'
import { CookiesProvider } from 'react-cookie'
import { appWithTranslation } from 'next-i18next'

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <CookiesProvider>
      <Head>
        <title>{FULL_NAME}</title>
      </Head>
      <Component {...pageProps} />
    </CookiesProvider>
  )
}

export default appWithTranslation(MyApp)
