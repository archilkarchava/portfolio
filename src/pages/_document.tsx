import Document, {
  DocumentContext,
  Head,
  Html,
  Main,
  NextScript,
} from 'next/document'

class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html lang="en">
        <Head>
          <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon" />
          <meta
            name="theme-color"
            media="(prefers-color-scheme: light)"
            content="#ffffff"
          />
          <meta
            name="theme-color"
            media="(prefers-color-scheme: dark)"
            content="#000000"
          />
          <meta name="description" content="Portfolio" />
          <meta
            name="keywords"
            content="Software Engineer, portfolio, frontend, backend, React, React Native, NestJS, nest, next, nextjs, tailwind, html, css, JavaScript, TypeScript, Node, nodejs, go, golang"
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument
