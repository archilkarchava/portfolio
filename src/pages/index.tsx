import { ApolloClient, gql, InMemoryCache } from '@apollo/client'
import GithubIcon from '@assets/github.svg'
import { GetStaticProps } from 'next'
import React from 'react'

const contactEmail = process.env.NEXT_PUBLIC_CONTACT_EMAIL ?? 'hello@archil.dev'
const githubLogin = process.env.NEXT_PUBLIC_GITHUB_LOGIN ?? 'archilkarchava'

interface Props {
  pinnedRepositories: {
    __typename: 'Repository'
    id: string
    name: string
    descriptionHTML: string
    url: string
    homepageUrl: string
  }[]
}

export const Home: React.FC<Props> = ({ pinnedRepositories }) => {
  return (
    <div className="flex items-center justify-center w-full h-full">
      <div className="max-w-4xl p-5 m-auto">
        <div>
          <h1 className="text-4xl font-bold leading-tight sm:text-6xl">
            Archil Karchava
          </h1>
          <div className="flex items-center mt-1 text-2xl sm:mt-3 sm:text-4xl">
            <a href={`mailto:${contactEmail}`} className="mr-4">
              {contactEmail}
            </a>
            <a
              title="My Github"
              href={`https://github.com/${githubLogin}`}
              target="_blank"
              rel="noopener noreferrer"
              className="text-current hover:opacity-70"
            >
              <GithubIcon className="w-6 h-6 sm:w-9 sm:h-9" />
            </a>
          </div>
        </div>
        <div className="my-8">
          <h1 className="mb-2 text-2xl sm:text-4xl">Some of my projects:</h1>
          <div className="flex flex-wrap -m-2">
            {pinnedRepositories.map((repo) => (
              <div
                key={repo.id}
                className="flex-grow w-full p-4 m-2 border border-gray-300 rounded-lg md:w-5/12 dark:border-gray-700"
              >
                <div>
                  <div className="flex flex-row flex-grow">
                    {repo.homepageUrl && (
                      <div className="flex flex-col items-start mr-2 text-lg font-semibold">
                        <span>website: </span>
                        <span>code: </span>
                      </div>
                    )}
                    <div className="flex flex-col flex-grow text-lg font-bold">
                      {repo.homepageUrl && (
                        <div>
                          <a
                            href={repo.homepageUrl}
                            target="_blank"
                            rel="noopener noreferrer"
                          >
                            <span>{repo.homepageUrl}</span>
                          </a>
                        </div>
                      )}
                      <a
                        href={repo.url}
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        <span>{repo.name}</span>
                      </a>
                    </div>
                  </div>
                  <div
                    className="text-base break-words"
                    dangerouslySetInnerHTML={{ __html: repo.descriptionHTML }}
                  />
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}

export const getStaticProps: GetStaticProps<Props> = async () => {
  const GITHUB_GRAPHQL_API_ENDPOINT =
    process.env.GITHUB_GRAPHQL_API_ENDPOINT ?? 'https://api.github.com/graphql'
  const token = process.env.GITHUB_TOKEN
  const client = new ApolloClient({
    uri: GITHUB_GRAPHQL_API_ENDPOINT,
    headers: {
      Authorization: token ? `bearer ${token}` : null,
    },
    cache: new InMemoryCache(),
  })

  let data: {
    repositoryOwner: {
      pinnedItems: {
        edges: {
          node: Props['pinnedRepositories'][number]
        }[]
      }
    }
  }
  try {
    const res = await client.query({
      query: gql`
        query PinnedRepositories($login: String!) {
          repositoryOwner(login: $login) {
            ... on User {
              pinnedItems(first: 6) {
                edges {
                  node {
                    ... on Repository {
                      id
                      name
                      descriptionHTML
                      url
                      homepageUrl
                    }
                  }
                }
              }
            }
          }
        }
      `,
      variables: { login: githubLogin },
    })
    data = res.data
  } catch (error) {
    throw new Error(error)
  }

  const pinnedRepositories = data?.repositoryOwner.pinnedItems.edges.map(
    (edge) => edge.node
  )

  return {
    props: {
      pinnedRepositories,
    },
    revalidate: 1,
  }
}

export default Home
