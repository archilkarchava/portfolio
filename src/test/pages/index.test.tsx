import React from 'react'
import { Home } from '../../pages/index'
import { render } from '@/test/testUtils'

describe('Home page', () => {
  const pinnedRepositories = [
    {
      id: 'MDEwOlJlcG9zaXRvcnkyNjQwMDQyOTU=',
      name: 'astragal-gatsby',
      descriptionHTML:
        '<div>E-commerce website built using Gatsby (React, TypeScript) and serverless.</div>',
      url: 'https://github.com/archilkarchava/astragal-gatsby',
      homepageUrl: 'https://astragal74.ru',
    },
    {
      id: 'MDEwOlJlcG9zaXRvcnkyNjkyNjEyNzU=',
      name: 'astragal-sanity-studio',
      descriptionHTML:
        '<div>Admin panel for <a href="https://astragal74.ru" rel="nofollow">https://astragal74.ru</a>\n' +
        '</div>',
      url: 'https://github.com/archilkarchava/astragal-sanity-studio',
      homepageUrl: '',
    },
    {
      id: 'MDEwOlJlcG9zaXRvcnkyMTY4ODU0MDU=',
      name: 'furnitur',
      descriptionHTML:
        '<div>Furnitur is a cross-platform (IOS, Android) e-commerce application built using Flutter framework and Firebase backend.</div>',
      url: 'https://github.com/archilkarchava/furnitur',
      homepageUrl: '',
    },
    {
      id: 'MDEwOlJlcG9zaXRvcnkyMTc2NjAyMDY=',
      name: 'furnitur-firebase',
      descriptionHTML: '<div>Firebase backend for Furnitur app.</div>',
      url: 'https://github.com/archilkarchava/furnitur-firebase',
      homepageUrl: null,
    },
  ]
  const HomeComponent = (
    <Home
      name="Ivan Ivanov"
      email="test@mail.com"
      pinnedRepositories={pinnedRepositories}
    />
  )

  it('matches snapshot', () => {
    const { asFragment } = render(HomeComponent, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
