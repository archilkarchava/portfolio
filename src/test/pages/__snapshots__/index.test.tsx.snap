// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    __source="[object Object]"
    class="flex items-center justify-center w-full h-full"
  >
    <div
      __source="[object Object]"
      class="max-w-4xl p-5 m-auto"
    >
      <div
        __source="[object Object]"
      >
        <h1
          __source="[object Object]"
          class="text-4xl font-bold leading-tight sm:text-6xl"
        >
          Ivan Ivanov
        </h1>
        <div
          __source="[object Object]"
          class="flex items-center mt-1 text-2xl sm:mt-3 sm:text-4xl"
        >
          <a
            __source="[object Object]"
            class="mr-4"
            href="mailto:test@mail.com"
          >
            test@mail.com
          </a>
          <a
            __source="[object Object]"
            class="text-current hover:opacity-70"
            href="https://github.com/archilkarchava"
            rel="noopener noreferrer"
            target="_blank"
            title="My Github"
          >
            <test-file-stub
              __source="[object Object]"
              class="w-6 h-6 sm:w-9 sm:h-9"
            />
          </a>
        </div>
      </div>
      <div
        __source="[object Object]"
        class="my-8"
      >
        <h1
          __source="[object Object]"
          class="mb-2 text-2xl sm:text-3xl"
        >
          Some of my projects:
        </h1>
        <div
          __source="[object Object]"
          class="flex flex-wrap -m-2"
        >
          <div
            __source="[object Object]"
            class="flex-grow w-full p-4 m-2 border border-gray-300 rounded-lg md:w-5/12 dark:border-gray-700"
          >
            <div
              __source="[object Object]"
            >
              <div
                __source="[object Object]"
                class="flex flex-row flex-grow"
              >
                <div
                  __source="[object Object]"
                  class="flex flex-col items-start mr-2 text-lg font-semibold"
                >
                  <span
                    __source="[object Object]"
                  >
                    website: 
                  </span>
                  <span
                    __source="[object Object]"
                  >
                    code: 
                  </span>
                </div>
                <div
                  __source="[object Object]"
                  class="flex flex-col flex-grow text-lg font-bold"
                >
                  <div
                    __source="[object Object]"
                  >
                    <a
                      __source="[object Object]"
                      href="https://astragal74.ru"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <span
                        __source="[object Object]"
                      >
                        https://astragal74.ru
                      </span>
                    </a>
                  </div>
                  <a
                    __source="[object Object]"
                    href="https://github.com/archilkarchava/astragal-gatsby"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      __source="[object Object]"
                    >
                      astragal-gatsby
                    </span>
                  </a>
                </div>
              </div>
              <div
                __source="[object Object]"
                class="text-base break-words"
              >
                <div>
                  E-commerce website built using Gatsby (React, TypeScript) and serverless.
                </div>
              </div>
            </div>
          </div>
          <div
            __source="[object Object]"
            class="flex-grow w-full p-4 m-2 border border-gray-300 rounded-lg md:w-5/12 dark:border-gray-700"
          >
            <div
              __source="[object Object]"
            >
              <div
                __source="[object Object]"
                class="flex flex-row flex-grow"
              >
                <div
                  __source="[object Object]"
                  class="flex flex-col flex-grow text-lg font-bold"
                >
                  <a
                    __source="[object Object]"
                    href="https://github.com/archilkarchava/astragal-sanity-studio"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      __source="[object Object]"
                    >
                      astragal-sanity-studio
                    </span>
                  </a>
                </div>
              </div>
              <div
                __source="[object Object]"
                class="text-base break-words"
              >
                <div>
                  Admin panel for 
                  <a
                    href="https://astragal74.ru"
                    rel="nofollow"
                  >
                    https://astragal74.ru
                  </a>
                  

                </div>
              </div>
            </div>
          </div>
          <div
            __source="[object Object]"
            class="flex-grow w-full p-4 m-2 border border-gray-300 rounded-lg md:w-5/12 dark:border-gray-700"
          >
            <div
              __source="[object Object]"
            >
              <div
                __source="[object Object]"
                class="flex flex-row flex-grow"
              >
                <div
                  __source="[object Object]"
                  class="flex flex-col flex-grow text-lg font-bold"
                >
                  <a
                    __source="[object Object]"
                    href="https://github.com/archilkarchava/furnitur"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      __source="[object Object]"
                    >
                      furnitur
                    </span>
                  </a>
                </div>
              </div>
              <div
                __source="[object Object]"
                class="text-base break-words"
              >
                <div>
                  Furnitur is a cross-platform (IOS, Android) e-commerce application built using Flutter framework and Firebase backend.
                </div>
              </div>
            </div>
          </div>
          <div
            __source="[object Object]"
            class="flex-grow w-full p-4 m-2 border border-gray-300 rounded-lg md:w-5/12 dark:border-gray-700"
          >
            <div
              __source="[object Object]"
            >
              <div
                __source="[object Object]"
                class="flex flex-row flex-grow"
              >
                <div
                  __source="[object Object]"
                  class="flex flex-col flex-grow text-lg font-bold"
                >
                  <a
                    __source="[object Object]"
                    href="https://github.com/archilkarchava/furnitur-firebase"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      __source="[object Object]"
                    >
                      furnitur-firebase
                    </span>
                  </a>
                </div>
              </div>
              <div
                __source="[object Object]"
                class="text-base break-words"
              >
                <div>
                  Firebase backend for Furnitur app.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
